lib/std/zon/stringify.zig

```diff
@@ -0,0 +1,2088 @@
+                                const default_field_value: *const @TypeOf(field_value) = @ptrCast(
+                                    @alignCast(default_field_value_opaque),
+                                );
+                                if (std.meta.eql(field_value, default_field_value.*)) {
+                                    skip.* = true;
+                                    fields -= 1;
+                                }
+                            }
+                        }
+                        break :b .{ fields, skipped };
+                    };
+
+                    // Emit those fields
+                    var container = try self.startStruct(
+                        .{ .whitespace_style = .{ .fields = fields } },
+                    );
+                    inline for (@"struct".fields, skipped) |field_info, skip| {
+                        if (!skip) {
+                            try container.fieldArbitraryDepth(
+                                field_info.name,
+                                @field(val, field_info.name),
+                                options,
+                            );
+                        }
+                    }
+                    try container.finish();
+                },
+                .@"union" => |@"union"| if (@"union".tag_type == null) {
+                    @compileError(@typeName(@TypeOf(val)) ++ ": cannot stringify untagged unions");
+                } else {
+                    var container = try self.startStruct(.{ .whitespace_style = .{ .fields = 1 } });
+                    switch (val) {
+                        inline else => |pl, tag| try container.fieldArbitraryDepth(
+                            @tagName(tag),
+                            pl,
+                            options,
+                        ),
+                    }
+                    try container.finish();
+                },
+                .optional => if (val) |inner| {
+                    try self.valueArbitraryDepth(inner, options);
+                } else {
+                    try self.writer.writeAll("null");
+                },
+
+                else => @compileError(@typeName(@TypeOf(val)) ++ ": cannot stringify this type"),
+            }
+        }
+
+        /// Serialize an integer.
+        pub fn int(self: *Self, val: anytype) Writer.Error!void {
+            try std.fmt.formatInt(val, 10, .lower, .{}, self.writer);
+        }
+
+        /// Serialize a float.
+        pub fn float(self: *Self, val: anytype) Writer.Error!void {
+            switch (@typeInfo(@TypeOf(val))) {
+                .float, .comptime_float => if (std.math.isNan(val)) {
+                    return self.writer.writeAll("nan");
+                } else if (@as(f128, val) == std.math.inf(f128)) {
+                    return self.writer.writeAll("inf");
+                } else if (@as(f128, val) == -std.math.inf(f128)) {
+                    return self.writer.writeAll("-inf");
```

Comment:
```suggestion
                } else if (std.math.isPositiveInf(val)) {
                    return self.writer.writeAll("inf");
                } else if (std.math.isNegativeInf(val)) {
                    return self.writer.writeAll("-inf");
```

Reply:
Unless this has some weirdness with `comptime_float`?

Reply:
Yeah it's cause of `comptime_float`. That being said I just moved the comptime float print to a separate branch anyway, may as well make it a bit easier to read.

Reply:
Makes sense -- that path will become unnecessary once #21205 is implemented.